src/routes/auth/index.ts

```diff
@@ -1,142 +1,288 @@
 
-	const isValid = await validatePassword(parsedBody.password, storedPassword);
-	if (!isValid) {
-		return context.json<StatusResponse>({ message: genericSignInError }, StatusCodes.UNAUTHORIZED);
+		return context.json({ message: 'Sign in successful' } satisfies StatusResponse, StatusCodes.CREATED);
 	}
+);
 
-	const sessionToken = crypto.randomUUID();
-	const hoursOffset = 24;
-	const tokenExpiryISO = addHours(new Date(), hoursOffset).toISOString();
-	const sessionDataObject: SessionData = {
-		id: profileId,
-		email: parsedBody.email,
-		access: accessLevel,
-		expiry: tokenExpiryISO
-	};
-	const sessionData = JSON.stringify(sessionDataObject);
-	await context.env.SESSION_TOKENS.put(sessionToken, sessionData);
-
-	context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiryISO}; Path=/;`);
-
-	return context.json({ message: 'Successfully signed in' });
+// Protected auth routes
+export const protectedAuthRoutes = new OpenAPIHono<EnvironmentBindings>({
+	defaultHook: statusResponseFormatter
 });
+
+protectedAuthRoutes.openapi(
+	createRoute({
+		method: 'get',
+		path: '/heartbeat',
+		operationId: 'getHeartbeat',
+		summary: 'Check if authenticated and get name, avatar and access',
+		descrition: 'Every protected UI page will make a heartbeat check and if successful will receive name, avatar and access in order to generate custsom content',
+		tags: ['heartbeat'],
+		responses: {
+			[StatusCodes.NOT_FOUND]: {
+				description: 'Internal error getting profile that should exist',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.UNAUTHORIZED]: {
+				description: 'No cookies found, invalid or missing token, invalid session or session expired. ',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.OKAY]: {
+				description: 'Heartbeat succesful. User authenticted and name, avatar and access returned to Client.',
+				content: { 'application/json': { schema: HeartbeatResponseSchema } }
+			}
+		},
+		middleware: [authorizeVolunteer] as const
+	}),
+	async (context) => {
+		const sessionData = context.get('session');
+
+		if (!sessionData) {
+			return context.json({ message: 'Invalid session' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
+		}
+
+		const profile = await getProfileById(context.env.database, sessionData.id);
+
+		if (!profile) {
+			return context.json({ message: 'Internal error getting profile that should exist' } satisfies StatusResponse, StatusCodes.NOT_FOUND);
+		}
+		const { name } = profile;
+		const { access } = sessionData;
+
+		// STUB: Avatar upload and resource under construction
+		const avatar = 'https://gravatar.com/avatar/f8eb6ba9cc4ad24f3b79897a8596ee90?s=400&d=robohash&r=x';
+
+		return context.json({ message: 'Heartbeat succesful', access, name, avatar } satisfies HeartbeatResponse, StatusCodes.OKAY);
```

Comment:
question: does the front-end need a `Heartbeat successful` message? Or is it enough to return `access`, `name`, `avatar`?

Reply:
Agreed.  Removed.